jujugui/static/gui/src/app/init/component-renderers-mixin.js

```diff
@@ -0,0 +1,1080 @@
+/* Copyright (C) 2017 Canonical Ltd. */
+'use strict';
+
+const yui = window.yui;
+
+const ComponentRenderersMixin = (superclass) => class extends superclass {
```

Comment:
A possible alternative idea is, instead of using a mixin for the renderer, defining a renderer class that gets instantiated passing the required objects (like the db, the topology etc). This class has methods for rendering stuff, like userProfile etc. This way the required attributes are explicit because they must be passed to the constructor. The JujuApp can then just do something like:
```javascript
this._render = new Renderer(this.db, this.topology...);
this._render.userProfile(...);
this._render.components();
```
What do you think?